framework/AGGRenderer/CPPAGGRenderer/CPPAGGRenderer.cpp

```diff
@@ -401,7 +427,7 @@ namespace CPPAGGRenderer{
       std::vector<unsigned char> image(buffer, buffer + (frame_width*frame_height*3));
       write_png(image, frame_width, frame_height, file_png);
 
-      // delete[] buffer;
+      delete[] buffer;
```

Comment:
By deleting the buffer here, will that lead to a problem if someone tries to save the same plot to disk multiple times? The buffer only appears to be initialized on creation of the Plot at a certain size.

Reply:
@BradLarson yes this might happen if the user draws the graph and tried to save it separately multiple times. I had only thought of the method drawAndSaveGraph(). 
What if I create a separate method that deletes the buffer and call it in the destructor of each Graph? But this would be AGG specific...

Reply:
Yeah, somehow you'll want to balance out the new / delete to avoid the leaks you were seeing earlier, while preventing overreleases. It would seem to make sense to delete the buffer on deallocation of the surrounding object, as well as before reshaping the Plot and requiring the creation of a new buffer. Not quite sure where that is, though.

Reply:
@BradLarson I apologise for the delayed response. I wasn't able to look into it yesterday. I'll get this done and merged today.